cmake/public/cuda.cmake

```diff
@@ -1,5 +1,13 @@
 # ---[ cuda
 
+# Check if we need to use the new FindCUDA functionalities, which are included
+# since 3.7.0. Also, we know that VS2017 needs the new FindCUDA functionality,
+# so we will simply enable it for the whole Windows build.
+if (MSVC OR ${CMAKE_VERSION} VERSION_LESS 3.7.0)
```

Comment:
This is wrong, we are tracking cmake MASTER; 3.7.0 is the wrong commit.

Reply:
All of the modifications in `Modules_CUDA_fix` exist in cmake master. So, hypothetically, once they do a release we could pin to that.

Though, I do agree it's less error prone if we just always include our own copy.